### PR TITLE
Remove --no-cache from buildah as no longer necessary

### PR DESCRIPTION
**Pull request by jimmyjones2**
Remove `--no-cache` as no longer necessary and prevents use of `--cache-from`/`--cache-to`

See #1229 

@vdemeester @holly-cummins 

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations][authoring]
- [ ] Includes [docs][docs] (if user facing)
- [ ] Includes [tests][tests] (for new tasks or changed functionality)
  - See the [end-to-end testing documentation][e2e] for guidance and CI details.
- [x] Meets the [Tekton contributor standards][contributor] (including functionality, content, code)
- [x] Commit messages follow [commit message best practices][commit]
- [x] Has a kind label. You can add one by adding a comment on this PR that
  contains `/kind <type>`. Valid types are bug, cleanup, design, documentation,
  feature, flake, misc, question, tep
- [ ] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [ ] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [ ] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [ ] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [ ] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [ ] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md) for more details._

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]: https://github.com/tektoncd/catalog/issues/413
[authoring]: https://github.com/tektoncd/catalog/blob/main/recommendations.md
[docs]: https://github.com/tektoncd/community/blob/master/standards.md#docs
[tests]: https://github.com/tektoncd/community/blob/master/standards.md#tests
[e2e]: https://github.com/tektoncd/catalog/blob/main/CONTRIBUTING.md#end-to-end-testing
[contributor]: https://github.com/tektoncd/community/blob/main/standards.md
[commit]: https://github.com/tektoncd/community/blob/master/standards.md#commit-messages

### CODE REVIEW

**Comment by chmouel** [comment]
this lgtm but is there any documentation or changelog upstream which documents that change of behaviour ? I could not find it anywhere..

**Comment by vinamra28** [comment]
> As it modify the behavior of the task, it should be in a new version of the `buildah` Task (`0.7`).
> 
> /cc @vinamra28

+1 to @vdemeester, let's bump the Task and also update the README with the changes so that users get to know what changed and what's the advantage. Maybe we can point to the GH issue or Buildah's documentation

File: task/buildah/0.6/buildah.yaml [hunk 1]
```diff
@@ -78,7 +78,7 @@ spec:
       # build the image (CERT_DIR_FLAG should be omitted if empty and BUILD_EXTRA_ARGS can contain multiple args)
       # shellcheck disable=SC2046,SC2086
       buildah ${CERT_DIR_FLAG} "--storage-driver=$(params.STORAGE_DRIVER)" bud $(params.BUILD_EXTRA_ARGS) \
-        "--format=$(params.FORMAT)" "--tls-verify=$(params.TLSVERIFY)" --no-cache \
+        "--format=$(params.FORMAT)" "--tls-verify=$(params.TLSVERIFY)" \
```
[the remark this answers is not in the document]
Comment (jimmyjones2) [reply task/buildah/0.6/buildah.yaml:81]: I don't see any benefit to ever setting --no-cache?

**Comment by jimmyjones2** [comment]
I'm happy to create a new version, but have we actually changed the functionality? I think we've fixed a bug which prevents users from enabling caching.

I suppose the worst that would happen is someone had enabled caching in BUILD_EXTRA_ARGS, which would have silently not worked, and now it would spring into life.

**Comment by jimmyjones2** [comment]
@vdemeester @vinamra28 Are you happy for this to be merged?

**Comment by vdemeester** [comment]
The "bump" the version when the behavior changed (a param, a flag, …), is kind-of to do a "tag are immutable" type of things. See versions as tag ; when we tag a commit, it's not very usual / common to force update the tag to something new (because then, when you look at your "checked-out" code from this tag, it might not be the correct one). This is to do the same for the catalog ; if I installed the task before this PR or after this PR, I will have different versions.

**Comment by jimmyjones2** [comment]
@vdemeester Created a new version, can you review please?

**Comment by jimmyjones2** [comment]
@chmouel @vinamra28 Can you review please?

**Review by vinamra28 — APPROVED** [review]
/ok-to-test
/lgtm

**Pull request merged**